lib/rust/ensogl/core/src/debug/monitor.rs

```diff
@@ -198,9 +198,64 @@ impl Drop for DomData {
 // === Monitor ===
 // ===============
 
-/// Implementation of the monitoring panel.
-#[derive(Debug)]
+/// Visual panel showing performance-related statistics.
+#[derive(Debug, Clone, CloneRef)]
 pub struct Monitor {
+    renderer: Rc<RefCell<Renderer>>,
+}
+
+impl Default for Monitor {
+    fn default() -> Self {
+        let mut renderer = Renderer::new();
+        renderer.add::<FrameTime>();
+        renderer.add::<Fps>();
+        renderer.add::<WasmMemory>();
+        renderer.add::<GpuMemoryUsage>();
+        renderer.add::<DrawCallCount>();
+        renderer.add::<DataUploadCount>();
+        renderer.add::<DataUploadSize>();
+        renderer.add::<BufferCount>();
+        renderer.add::<SymbolCount>();
+        renderer.add::<ShaderCount>();
+        renderer.add::<ShaderCompileCount>();
+        renderer.add::<SpriteSystemCount>();
+        renderer.add::<SpriteCount>();
+        Self { renderer: Rc::new(RefCell::new(renderer)) }
+    }
+}
+
+impl Monitor {
+    /// Constructor.
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Draw the monitor and update its graphs based on the provided stats data.
+    pub fn draw(&self, stats: &StatsData) {
+        let mut renderer = self.renderer.borrow_mut();
```

Comment:
This is a bad pattern to borrow_mut for the whole scope. Someone will refactor this code in the future, put her borrows in the lines below and it will blow up. Better way of doing it is the way that function `toggle` is implemented - borrow once and redirect the call to the borrowed structure.